lib/features/family/features/home_screen/presentation/pages/navigation_bar_home_screen.dart

```diff
@@ -200,7 +201,13 @@ class _NavigationBarHomeScreenState extends State<NavigationBarHomeScreen> {
   ) async {
     switch (custom) {
       case PreferredChurchDialog():
-        showPreferredChurchModal(context);
+        if (_isShowingPreferredChurch) {
```

Comment:
Wouldn't it be better to make this check part of showPreferredChurchModal? :)
The method itself should never trigger the modal twice.
The method can also reset the variable if needed (idk if thats needed, but it doesn't do that right now)

Reply:
That would be a good improvement indeed! Will do!